torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -275,6 +276,52 @@ class CudaKernelGenerator : private kir::IrVisitor {
   }
 
   void visit(const kir::UnaryOp* node) final {
+    bool is_vector_op = false;
+    size_t vector_word_size = 1;
+
+    if (node->out()->isA<kir::TensorIndex>()) {
+      auto ti = node->out()->as<kir::TensorIndex>();
+      for (auto id : ti->view()->fuserTv()->domain()->domain()) {
+        if (id->getParallelType() != ParallelType::Vectorize) {
```

Comment:
How do you plan to distinguish a vectorized read / write and the misaligned set operations?

I created a `UnaryOp::VectorizedSet` operation for this.

Reply:
I was trying to figure out why we needed to distinguish, can you explain in a bit more detail?

Reply:
The pre and post misaligned for-loops use a regular set operation.

The three for-loops share the same vectorization iterDomain. How can this distinguish them?

Reply:
Okay, understood. It can't, there's two parts we need. (1) is a separate type of vectorization on the TensorView to know which type of vectorization to use. This could be `Vectorize` and `EnforceVectorize`, or something similar on the `ParallelType`. This is similar to what was done with `Broadcast` Vs `StridedBroadcast` on `IterDomainType`.

This information does need to resolve here as well as, indexing for what you're talking about. I don't have a clean mechanism in mind to do this. Doing it on IterDomain would be quite annoying. Doing it on the ForLoop may be more reasonable. We have the vectorized for loop piped through the system and could mark it directly with "enforce" or normal vectorization. Indexing and codegen would need to process this information which yes, is different than what is done here.